be/src/exec/spill/block_reader.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "exec/spill/block_manager.h"
+#include "fmt/format.h"
+#include "util/slice.h"
+
+namespace starrocks::spill {
+
+Status BlockReader::read_fully(void* data, int64_t count) {
```

Comment:
why not use buffered input stream?

Reply:
> why not use buffered input stream?

BufferInputStream only perceives the concept of Chunk, and the buffer read here needs to be done at a lower level

Reply:
```
class SharedBufferedInputReadable {
      Status read() {
      // try to read from buffer.
      ...
      // else call readable->read
      }
}
```

Reply:
> ```
> class SharedBufferedInputReadable {
>       Status read() {
>       // try to read from buffer.
>       ...
>       // else call readable->read
>       }
> }
> ```

I don't see the benefit of adding this layer of abstraction. I think it's OK to let BlockReader do it.